qttypes/src/lib.rs

```diff
@@ -640,23 +641,105 @@ impl std::fmt::Debug for QString {
     }
 }
 
+// XXX: The Default, Clone and Copy traits are provided by cpp_class! but they won't have these
+// nice "Wrapper around ..." docs with links on them. How can we fix this?
 cpp_class!(
     /// Wrapper around [`QVariant`][class] class.
     ///
     /// [class]: https://doc.qt.io/qt-5/qvariant.html
     #[derive(PartialEq)]
     pub unsafe struct QVariant as "QVariant"
 );
+// methods are ordered as in the official documentation
 impl QVariant {
-    /// Wrapper around [`toByteArray()`][method] method.
+    /// Wrapper around [`canConvert(int targetTypeId)`][method] method.
     ///
-    /// [method]: https://doc.qt.io/qt-5/qvariant.html#toByteArray
-    pub fn to_qbytearray(&self) -> QByteArray {
-        cpp!(unsafe [self as "const QVariant*"] -> QByteArray as "QByteArray" {
-            return self->toByteArray();
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#canConvert
+    pub fn can_convert(&self, target_type_id: i32) -> bool {
+        cpp!(unsafe [self as "const QVariant*", target_type_id as "int"] -> bool as "bool" {
+            return self->canConvert(target_type_id);
+        })
+    }
+
+    // TODO: for this to work, we need QMetaType be available in qttypes package.
+    // TODO: It would create a dependency cycle at this point.
+    // /// Wrapper around [`canConvert()`][method] method.
+    // ///
+    // /// [method]: https://doc.qt.io/qt-5/qvariant.html#canConvert-1
+    // pub fn can_convert<T: QMetaType>(&self) -> bool {
+    //     self.can_convert(T::id())
+    // }
+
+    /// Wrapper around [`clear()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#clear
+    pub fn clear(&mut self) {
+        cpp!(unsafe [self as "QVariant*"] {
+            self->clear();
+        });
+    }
+
+    /// Wrapper around [`convert(int targetTypeId)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#convert
+    pub fn convert(&mut self, target_type_id: i32) -> bool {
+        cpp!(unsafe [self as "QVariant*", target_type_id as "int"] -> bool as "bool" {
+            return self->convert(target_type_id);
+        })
+    }
+
+    // TODO?: [static] template <typename Types> QVariant QVariant::fromStdVariant(const std::variant<Types...> &value)
```

Comment:
In general, we don't have to replicate the whole API, only the API that make sense.  This clearly make no sense in rust.
Same for the function that operates on the QMetaType which are better done by the QMetaType trait.

Reply:
agreed.